### PR TITLE
Fixed some Deprecation errors in SBT.

### PR DESCRIPTION
**Pull request by zarthross**
I refactored the build to remove some of the deprecation errors.

The only error I did not get to was this: 
```
build.sbt:120: warning: method fileToRef in object Reference is deprecated: Explicitly wrap the File in a call to RootProject.
  packagedArtifacts in file(".") := Map.empty,
                           ^
```
You guys might take your time looking over the bits I changed in `VersionedGhPages` because I'm not able to easily test your publishing code.

### CODE REVIEW

**Comment by zarthross** [comment]
Obviously, if you guys don't like the way I did anything, let me know.  I'll be happy to correct any issues.

**Comment by ChristopherDavenport** [comment]
Looks pretty nice on an initial look. As it's a lot of new code I will try to come around to confirm in depth shortly when I get a minute.

File: build.sbt [hunk 1]
```diff
@@ -1,4 +1,161 @@
...
+lazy val dontPublish = packagedArtifacts := Map.empty
+
+lazy val license = licenses in ThisBuild := Seq(
+  "Apache License, Version 2.0" -> url(
+    "http://www.apache.org/licenses/LICENSE-2.0.txt")
+)
+
+lazy val buildSettings = publishing ++
+  Seq(
+    scalaVersion := "2.12.3",
```
Comment (rossabaker) [inline build.sbt]: This could be 2.12.4 now.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import sbt._
+import Keys._
+
+object Dependencies {
+  lazy val http4sVersion = "0.17.1"
```
Comment (rossabaker) [inline project/Dependencies.scala]: This could be 0.17.5 now.

File: project/VersionedGhPages.scala [hunk 3]
```diff
@@ -10,34 +10,35 @@ import com.typesafe.sbt.git.GitRunner
...
-	def cleanSiteForRealz(dir: File, git: GitRunner, s: TaskStreams, apiVersion: (Int, Int)): Unit ={
-		val toClean = IO.listFiles(dir).collect {
-			case f if f.getName == "api" => new java.io.File(f, s"${apiVersion._1}.${apiVersion._2}")
-			case f if f.getName != ".git" && f.getName != "CNAME" => f
-		}.map(_.getAbsolutePath).toList
-		if (!toClean.isEmpty)
-			git(("rm" :: "-r" :: "-f" :: "--ignore-unmatch" :: toClean) :_*)(dir, s.log)
-		()
-	}
+  def cleanSiteForRealz(dir: File, git: GitRunner, s: TaskStreams, apiVersion: (Int, Int)): Unit = {
```
Comment (rossabaker) [inline project/VersionedGhPages.scala]: The latest sbt-ghpages could make all this nonsense go away, but that's probably out of scope for this request.

**Comment by jcranky** [comment]
@rossabaker I can deal with the conflicts in my fork, no worries there :)

**Comment by zarthross** [comment]
Updated the versions.  I also rebased the changes on the latest master.
(Also had to correct the author tags, they were a little messed up)

File: build.sbt [hunk 4]
```diff
@@ -1,4 +1,161 @@
...
+
+lazy val license = licenses in ThisBuild := Seq(
+  "Apache License, Version 2.0" -> url(
+    "http://www.apache.org/licenses/LICENSE-2.0.txt")
+)
+
+lazy val buildSettings = publishing ++
+  Seq(
+    scalaVersion := "2.12.4",
+    crossScalaVersions := Seq("2.12.3", "2.11.11"),
```
Comment (ChristopherDavenport) [inline build.sbt]: Missed a `2.12.4` here

Comment (rossabaker) [reply build.sbt]: Do we even need `crossScalaVersions` anymore?  The only thing it buys, I think, is local running with `+` prefix.  We let Travis handle the cross builds in our other repos.

Comment (zarthross) [reply build.sbt]: I think its useful to have.  It allows me to test my changes fully before I push them by running '+test'.
I'll get the version updated as well.

**Pull request merged**